module/applications/item/item-sheet.mjs

```diff
@@ -235,8 +235,7 @@ export default class ItemSheet5e extends ItemSheet {
     const baseIds = CONFIG.DND5E[`${type}Ids`];
     if ( baseIds === undefined ) return {};
 
-    const typeProperty = type === "armor" ? "armor.type" : `${type}Type`;
-    const baseType = foundry.utils.getProperty(this.item.system, typeProperty);
+    const baseType = this.item.system.type.value;
 
     const items = {};
     for ( const [name, id] of Object.entries(baseIds) ) {
```

Comment:
Line 243 below here will also need to change now that `typeProperty` has been removed:

```javascript
if ( baseType !== baseItem?.system.type?.value ) continue;
```